### PR TITLE
Adds "ca-dir" settings to configMap for eck-operator

### PR DESCRIPTION
**Pull request by jbouse**
Adds support for the `ca-dir` setting in the configMap as mentioned in #6435 and #6091.

### CODE REVIEW

File: deploy/eck-operator/values.yaml [hunk 1]
```diff
@@ -178,6 +178,9 @@ config:
   # caRotateBefore defines when to rotate a CA certificate that is due to expire.
   caRotateBefore: 24h
 
+  # caDir defines the directory containing a CA certificate (tls.crt) and its associated private key (tls.key) to be used for all managed resources.
+  caDir: ""
```
Comment (thbkrkr) [inline deploy/eck-operator/values.yaml:182]:
```suggestion
  # Setting this makes caRotateBefore and caValidity values ineffective.
  # caDir: ""
```

**Comment by thbkrkr** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by thbkrkr** [comment]
buildkite test this

**Review by pebrc — APPROVED** [review]
LGTM

**Comment by jbouse** [comment]
I had tested it locally with both `# caDir: ""` and `caDir: ""` and the behavior was the same so I'd left it uncommented as it was the default value.

**Comment by thbkrkr** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by thbkrkr** [comment]
buildkite test this

**Pull request merged**